homeassistant/components/google_generative_ai_conversation/entity.py

```diff
@@ -225,6 +223,18 @@ def _convert_content(
     if content.content:
         parts.append(Part.from_text(text=content.content))
 
+    if content.thinking_content:
+        parts.append(Part.from_text(text=content.thinking_content))
+        parts[-1].thought = True
+
+    if (
+        content.native
+        and isinstance(content.native, dict)
+        and content.native.get("domain") == DOMAIN
```

Comment:
I don't think that this is a good check. Why not just check if it's `agent_id` is equivalent?

Reply:
Otherwise, maybe we shuld make a dataclass for this integration that we can instance check.